src/coreclr/jit/morph.cpp

```diff
@@ -14534,6 +14534,38 @@ GenTree* Compiler::fgMorphMultiOp(GenTreeMultiOp* multiOp)
     if (opts.OptimizationEnabled() && multiOp->OperIs(GT_HWINTRINSIC))
     {
         GenTreeHWIntrinsic* hw = multiOp->AsHWIntrinsic();
+
+        // Move constant vectors from op1 to op2 for commutative and compare operations
+        // For now we only do it for zero vector
+        if (fgGlobalMorph && (HWIntrinsicInfo::lookupNumArgs(hw->GetHWIntrinsicId()) == 2) && hw->Op(1)->IsVectorZero())
+        {
+            bool swap = HWIntrinsicInfo::IsCommutative(hw->GetHWIntrinsicId());
+            if (!swap)
+            {
+                switch (hw->GetHWIntrinsicId())
```

Comment:
Why restrict this to equality?

Presumably if `HWIntrinsicInfo::IsCommutative` is `true` (could also be checked via `OperIsCommutative` I suppose), then this swapping is always legal.

Reply:
op_Equality/Inequaility aren't marked as commutative in the table

Reply:
I see, I wonder why and whether they can be marked as such.

Reply:
I'll try marking those as commutative once current spmi session finishes

Reply:
They should be. Likewise we should really handle and understand cases like `GT` where swapping operands is fine if we also use `GenCondition::Swap`